pkg/monitoring/metrics/virt-controller/vmstats_collector_test.go

```diff
@@ -780,24 +831,42 @@ var _ = Describe("VM Stats Collector", func() {
 				},
 			}
 
-			metrics := CollectVmsVnicInfo([]*k6tv1.VirtualMachine{vm})
-			Expect(metrics).To(BeEmpty())
+			metrics := CollectVmsVnicInfo(vmWithClone(vm))
+			Expect(filterResultsByVM(metrics, "test-vm", "test-ns")).To(BeEmpty())
+			Expect(filterResultsByVM(metrics, "test-vm-2", "test-ns")).To(BeEmpty())
 		})
 	})
 })
 
-func expectDefaultCPUResourceRequests(crs []operatormetrics.CollectorResult) {
-	Expect(crs).To(HaveLen(3), "Expected 3 metrics")
-
-	Expect(crs[0].Metric.GetOpts().Name).To(ContainSubstring("kubevirt_vm_resource_requests"))
-	Expect(crs[0].Value).To(BeEquivalentTo(1))
-	Expect(crs[0].Labels).To(Equal([]string{"testvm", "test-ns", "cpu", "cores", "default"}))
+// Add helpers for two-VM testing
+func vmWithClone(vm *k6tv1.VirtualMachine) []*k6tv1.VirtualMachine {
+	vm2 := vm.DeepCopy()
+	if vm2.Name == "" {
+		vm2.Name = vm.Name + "-2"
+	} else {
+		vm2.Name = vm2.Name + "-2"
+	}
+	if vm2.Namespace == "" {
+		vm2.Namespace = vm.Namespace
+	}
+	return []*k6tv1.VirtualMachine{vm, vm2}
+}
 
-	Expect(crs[1].Metric.GetOpts().Name).To(ContainSubstring("kubevirt_vm_resource_requests"))
-	Expect(crs[1].Value).To(BeEquivalentTo(1))
-	Expect(crs[1].Labels).To(Equal([]string{"testvm", "test-ns", "cpu", "threads", "default"}))
+func filterResultsByVM(crs []operatormetrics.CollectorResult, name, namespace string) []operatormetrics.CollectorResult {
+	var out []operatormetrics.CollectorResult
+	for _, r := range crs {
+		if len(r.Labels) >= 2 && r.Labels[0] == name && r.Labels[1] == namespace {
+			out = append(out, r)
+		}
+	}
+	return out
+}
 
-	Expect(crs[2].Metric.GetOpts().Name).To(ContainSubstring("kubevirt_vm_resource_requests"))
-	Expect(crs[2].Value).To(BeEquivalentTo(1))
-	Expect(crs[2].Labels).To(Equal([]string{"testvm", "test-ns", "cpu", "sockets", "default"}))
+func expectForBothVMs(
+	results []operatormetrics.CollectorResult,
+	baseName, namespace string,
+	expectVM func([]operatormetrics.CollectorResult, string, string),
+) {
```

Comment:
nit: ```suggestion
func expectVMs(
	results []operatormetrics.CollectorResult,
	namespace string,
	names []string,
	expectVM func([]operatormetrics.CollectorResult, string, string),
) {
```

Reply:
fixed the names but I think we should leave +2 since it's more clear and save the need to send 2 names